libgnucash/engine/Transaction.c

```diff
@@ -2197,6 +2197,8 @@ xaccTransSetDocLink (Transaction *trans, const char *doclink)
     if (!trans || !doclink || !g_strcmp0 (doclink, trans->docLink)) return;
 
     xaccTransBeginEdit(trans);
+    trans->docLink_from_kvp = FALSE;
+    g_free (trans->docLink);
     if (g_strcmp0 (doclink, "") == 0)
     {
```

Comment:
That doesn't work the way you want. `g_strcmp0(nullptr, "")` will return `-1`. Do `if (doclink && !*doclink)`.